litestar/_asgi/routing_trie/mapping.py

```diff
@@ -189,33 +189,36 @@ def build_route_middleware_stack(
     from litestar.middleware.response_cache import ResponseCacheMiddleware
     from litestar.routes import HTTPRoute
 
-    # we wrap the route.handle method in the ExceptionHandlerMiddleware
-    asgi_handler = wrap_in_exception_handler(
-        app=route.handle,  # type: ignore[arg-type]
-        exception_handlers=route_handler.resolve_exception_handlers(),
+    asgi_handler: ASGIApp = route.handle  # type: ignore[assignment]
+    handler_middleware = route_handler.resolve_middleware()
+    has_cached_route = isinstance(route, HTTPRoute) and any(r.cache for r in route.route_handlers)
+    has_middleware = (
+        app.csrf_config or app.compression_config or has_cached_route or app.allowed_hosts or handler_middleware
     )
 
-    if app.csrf_config:
-        asgi_handler = CSRFMiddleware(app=asgi_handler, config=app.csrf_config)
+    if has_middleware:
+        # if there is middleware, they may wrap the `send()` coro, so we need to wrap the handler in
```

Comment:
```suggestion
        # if there is middleware, they may wrap the `send()` callable, so we need to wrap the handler in
```

Nit but it's not really a coroutine just yet but a coroutine function :eyes:

Reply:
This whole comment was pretty poorly written, so I rewrote it and included this adjustement.